contracts/WETH10.sol

```diff
@@ -37,52 +37,64 @@ contract WETH10 {
                 keccak256(bytes("1")),
                 chainId,
                 address(this)));
+
+        // Trick to prevent transfers to the contract address without adding additional gas costs
+        balance[address(this)] = uint256(-1);
```

Comment:
This is quite a neat trick. Please use `type(uint256).max` instead of `uint256(-1)`, though.